event-publisher-spi/src/main/scala/uk/gov/nationalarchives/eventpublisherspi/helpers/EventUtils.scala

```diff
@@ -0,0 +1,22 @@
+package uk.gov.nationalarchives.eventpublisherspi.helpers
+
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder, parser}
+import org.keycloak.events.admin.{AdminEvent, OperationType, ResourceType}
+import uk.gov.nationalarchives.eventpublisherspi.EventPublisherProvider.{EventDetails, RoleMappingRepresentation}
+
+object EventUtils {
+  implicit val roleMappingRepresentationDecoder: Decoder[RoleMappingRepresentation] = deriveDecoder[RoleMappingRepresentation]
```

Comment:
Do you need the `deriveDecoder` implicit here? You're decoding a case class so can you just use the auto magic decoder and import `import io.circe.generic.auto._` instead? I tried it and it compiles ok.